wagtail/admin/templates/wagtailadmin/pages/confirm_delete.html

```diff
@@ -68,6 +68,11 @@
 
         <form action="{% url 'wagtailadmin_pages:delete' page.id %}" method="POST">
             {% csrf_token %}
+            {% if confirm_before_delete %}
+                <input type="checkbox" required>
+                {% trans "This will also delete all the child pages." %}
```

Comment:
This text doesn't really make it clear that this is a confirmation checkbox - it sounds more like "tick this box if you want to delete the child pages, otherwise leave it unchecked and it'll just delete the parent". I'd suggest the text:

"Yes, I confirm that I wish to delete `<N>` pages."

However, I think it would be better to make this a text input where the admin needs to actually type something, like Github uses:
![Screenshot 2022-06-20 at 14 51 59](https://user-images.githubusercontent.com/85097/174616750-b0bc6fe5-2c02-4a3c-90f2-f0949990bdfc.png)

The aim here is to give the user a task that they need to actively think about, instead of doing it "on autopilot" - if a user is careless enough to click the "Delete" button without reading the warning, they'll probably also prepared to tick a checkbox without reading it too. Maybe we can use [WAGTAIL_SITE_NAME](https://docs.wagtail.org/en/stable/reference/settings.html#wagtail-site-name) for the text they need to type (or maybe @wagtail/ux-design has some thoughts?)

Reply:
Ya, this could also be a solution. I will try to implement this approach as well.

Reply:
@gasman I think If we want to implement it this way then we don't need to use `WAGTAILADMIN_UNSAFE_PAGE_DELETION_LIMIT` we will just every time ask the admin to type the site name to confirm. Let me know what you think.

Reply:
No, the whole point of this change is that the admin is forced to do something different for large-scale deletions than for routine deletions of just a few pages, so that they're alerted to the fact that this is a more destructive change than usual.

If we make them type the site name on every single deletion, then A) they'll get annoyed very quickly, and B) they'll get used to typing it without thinking, and we're back to the original problem of people deleting things by mistake.

Reply:
:sweat_smile: ya so the point is we just need to alert the admin that he is going to do something destructive that is not going to be restored.

Reply:
![delete-confirm-1](https://user-images.githubusercontent.com/69707565/174830645-212fa549-5ca2-42ea-82b6-fea86837c35f.png)

![delete-confirm-2](https://user-images.githubusercontent.com/69707565/174827736-a6fd8c55-a218-497c-ab3e-51fc7f28f2ae.png)

@gasman let me know which one we can use or if have some other suggestions :)